scripts/main.component.ts

```diff
@@ -20,18 +21,36 @@ import { REPLComponent, REPLDelegate } from "./repl.component"
   templateUrl: "../html/main.component.html",
   providers: [MenuService, PluginService]
 })
-export class MainComponent implements OnInit, MenuDelegate, REPLDelegate {
+export class MainComponent implements OnInit, MenuDelegate, REPLDelegate, EditorDelegate {
   constructor(private menu: MenuService, private pluginService: PluginService) {}
+  
+  selectedElements : Set<DrawableGraph | DrawableNode | DrawableEdge> =
+    new Set<DrawableGraph | DrawableNode | DrawableEdge>();
+  selectElement(element : DrawableGraph | DrawableNode | DrawableEdge) {
```

Comment:
The editor component originally did the bookkeeping for selected items. I'm not sure that I like the idea of having that logic in the component itself, but I can't think of a better place to put it. We can put that stuff back into the component and then have the component expose an event handler field `selectionchanged` or something that the outside world can bind to. I had a note about it in one of the comments.